test/object-store/sync/app.cpp

```diff
@@ -3232,16 +3233,23 @@ TEST_CASE("app: sync integration", "[sync][pbs][app][baas]") {
         test_session.reopen(false);
 
         // Open the realm with the cached user
-        {
-            auto user = test_session.app()->current_user();
-            REQUIRE(user);
-            // Verify no transport calls have been made at this point since the
-            // app was re-opened.
-            REQUIRE(request_count == 0);
+        int num_realms = GENERATE(1, 3);
+        auto user = test_session.app()->current_user();
+        REQUIRE(user);
+        // Verify no transport calls have been made at this point since the
+        // app was re-opened.
+        REQUIRE(request_count == 0);
+        SyncTestFile r_config(user, partition, schema);
+
+        std::vector<SharedRealm> realms;
+        int i;
```

Comment:
any reason why `i` is declared outside of the loop?

Reply:
I was thinking to use the same var for both loops, but it's probably more efficient to declare it in each loop

Reply:
But there's only one loop that uses a non-range-based for loop?

Reply:
Oh yeah - there used to be two until I updated the second loop.